opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/export/__init__.py

```diff
@@ -322,10 +322,14 @@ def collect(self, timeout_millis: float = 10_000) -> None:
             )
             return
 
-        self._receive_metrics(
-            self._collect(self, timeout_millis=timeout_millis),
-            timeout_millis=timeout_millis,
-        )
+        metrics = self._collect(self, timeout_millis=timeout_millis)
+
+        if metrics is not None:
```

Comment:
@aabmass @lzchen In the previous SIG it was suggested that we produce zero-valued data points when using a Synchronous instrument and delta temporality if no measurements were produced between collection cycles. We should not do that, the spec [says](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md):

> For synchronous instruments with Delta aggregation temporality, [MetricReader.Collect](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#collect) MUST only receive data points with measurements recorded since the previous collection.

@jmacd pointed me to this part of the spec and also mentioned that not calling the exporter in this situation is actually and advantage of delta temporality, which makes sense in terms of resource usage.

Right now we are producing data points with synchronous instruments and delta temporality (which is a bug, mentioned it in #3418). I'll fix that but we also _need_ this PR changes because once that is fixed, we will end up having `None` in the `metrics` variable here and we need to handle that situation.

Reply:
That makes sense 👍 one other question, do we keep the oldest start time that had data in this case? I believe we should. Here's a scenario:

- 1st collection at t1, a delta counter has value 5
- 2nd collection at t2, the counter has None as no observations have been made
- 3rd collection at t3, the counter has value 2.

Does the 3rd collection send the time/start_time as `(t1, t3)` or `(t2, t3)`?